### PR TITLE
Add UTS namespace

### PR DESCRIPTION
**Pull request by cclerget**
**Description of the Pull Request (PR):**

It adds UTS namespace. After some tests to launch /sbin/init as root within an daemon instance show init (systemd) modifying "host" hostname based on container's /etc/hostname. So it could be useful and required for this use case. UTS is enabled by default even if it runs as root or normal user.

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [x] I have tested this PR locally with a `make test`
- [x] This PR is NOT against the project's master branch
- [x] I have added myself as a contributor to the [Author's file](https://github.com/singularityware/singularity/blob/master/AUTHORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularityware-admin

### CODE REVIEW

File: src/lib/runtime/ns/uts/uts.c [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ * your rights to use or distribute this software.
+ * 
+ * NOTICE.  This Software was developed under funding from the U.S. Department of
+ * Energy and the U.S. Government consequently retains certain rights. As such,
+ * the U.S. Government has been granted for itself and others acting on its
+ * behalf a paid-up, nonexclusive, irrevocable, worldwide license in the Software
+ * to reproduce, distribute copies to the public, prepare derivative works, and
+ * perform publicly and display publicly, and to permit other to do so. 
+ * 
+*/
```
Comment (bauerm97) [inline src/lib/runtime/ns/uts/uts.c]: This should be changed to the new copyright header format

File: src/lib/runtime/ns/uts/uts.h [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * your rights to use or distribute this software.
+ * 
+ * NOTICE.  This Software was developed under funding from the U.S. Department of
+ * Energy and the U.S. Government consequently retains certain rights. As such,
+ * the U.S. Government has been granted for itself and others acting on its
+ * behalf a paid-up, nonexclusive, irrevocable, worldwide license in the Software
+ * to reproduce, distribute copies to the public, prepare derivative works, and
+ * perform publicly and display publicly, and to permit other to do so. 
+ * 
+*/
```
Comment (bauerm97) [inline src/lib/runtime/ns/uts/uts.h]: As stated in the source file, this copyright header should be updated to the new version

**Review by bauerm97 — CHANGES_REQUESTED** [review]
Everything looks good to me, besides the copyright headers. I'm wondering if we always want to be invoking the UTS namespace or if it should only be invoked on the users request. If it should only be at user request, we should add a --uts option to the action_argparser.sh in libexec/cli/action_argparser.sh to allow the user to toggle this

**Comment by cclerget** [comment]
This namespace is required as long container is ran as root, so maybe a conditional statement if uid == 0 ?

**Comment by cclerget** [comment]
@bauerm97 up, finally enforce UTS namespace when container is run as root and add --uts option for user request

**Comment by cclerget** [comment]
Close PR, I'll merge it with later

**Pull request closed without merging**